### PR TITLE
ENH: np.angle: Remove unnecessary multiplication, and allow subclasses to pass through

### PR DESCRIPTION
**Pull request by eric-wieser**
…

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1339,18 +1339,18 @@ def angle(z, deg=0):
...
-        fact = 1.0
-    z = asarray(z)
-    if (issubclass(z.dtype.type, _nx.complexfloating)):
+    z = asanyarray(z)
+    if issubclass(z.dtype.type, _nx.complexfloating):
         zimag = z.imag
         zreal = z.real
     else:
         zimag = 0
         zreal = z
```
Comment (tylerjereddy) [inline numpy/lib/function_base.py]: perhaps worth noting that the code path where z isn't a subclass of `complexfloating` doesn't have test coverage, although that's probably not an issue with changes made here

**Review by tylerjereddy — COMMENTED** [review]
Looks like a net gain in efficiency for `angle`. One suggestion might be a unit test that ensures that when `z` is a subclass of `ndarray` (i.e., `np.matrix`) that it remains as such after `angle` is called on it. Maybe overkill, but nice to prevent regression after this improvement.

**Comment by tylerjereddy** [comment]
Actually, not sure you could easily test for that -- even the returned object is described as `ndarray` or scalar--is there a case where the subclass passthrough actually provides enhanced functionality?

**Comment by eric-wieser** [comment]
Test added for the parts that changed in this patch.

**Review by tylerjereddy — APPROVED** [review]
Looks solid now. Would you say it is pretty unlikely that users depended on `angle` returning the `ndarray` base class only beforehand? 

Not sure if the docstring for `Returns` should get a small update to reflect the increased flexibility in return type, perhaps in another small future PR.

**Comment by eric-wieser** [comment]
Added a comment about that

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -1339,18 +1342,18 @@ def angle(z, deg=0):
...
-    return arctan2(zimag, zreal) * fact
+
+    a = arctan2(zimag, zreal)
+    if deg:
+        a *= 180/pi
```
Comment (charris) [inline numpy/lib/function_base.py]: Maybe `np.rad2deg`? Probably no faster.

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**